x/ibc/04-channel/keeper/packet.go

```diff
@@ -294,20 +283,32 @@ func (k Keeper) ReceiveExecuted(
 
 		nextSequenceRecv++
 
-		// incrementng nextSequenceRecv and storing under this chain's channelEnd identifiers
+		// incrementing nextSequenceRecv and storing under this chain's channelEnd identifiers
 		// Since this is the receiving chain, our channelEnd is packet's destination port and channel
 		k.SetNextSequenceRecv(ctx, packet.GetDestPort(), packet.GetDestChannel(), nextSequenceRecv)
+	} else {
+		// For unordered channels we must set the receipt so it can be verified on the other side.
+		// This receipt does not contain any data, since the packet has not yet been processed,
+		// it's just a single store key set to an empty string to indicate that the packet has been received
+		_, found := k.GetPacketReceipt(ctx, packet.GetDestPort(), packet.GetDestChannel(), packet.GetSequence())
+		if found {
+			return sdkerrors.Wrapf(
+				types.ErrPacketReceived,
+				"destination port: %s, destination channel: %s, sequence: %d", packet.GetDestPort(), packet.GetDestChannel(), packet.GetSequence(),
+			)
+		}
+
+		k.SetPacketReceipt(ctx, packet.GetDestPort(), packet.GetDestChannel(), packet.GetSequence())
 	}
 
 	// log that a packet has been received & executed
-	k.Logger(ctx).Info(fmt.Sprintf("packet received & executed: %v", packet))
+	k.Logger(ctx).Info("packet received and executed", "packet", fmt.Sprintf("%v", packet))
```

Comment:
```suggestion
	k.Logger(ctx).Info("packet received", "packet", fmt.Sprintf("%v", packet))
```
I feel like executed might imply an acknowledgement?